### ISSUE TITLE
TypeError: Cannot read property 'range' of null

### ISSUE THREAD

**Issue by jayenashar** [babel/babel#13060]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**

```
TypeError: Cannot read property 'range' of null
Occurred while linting /home/jayen/docs/work/ridethewaverley/ride-the-waverley/src/util/writeresource.js:1
    at OffsetStorage.setDesiredOffset (/home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/rules/indent.js:341:45)
    at /home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/rules/indent.js:1397:29
    at Array.forEach (<anonymous>)
    at Object.TemplateLiteral [as listener] (/home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/rules/indent.js:1389:34)
    at /home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/rules/indent.js:1635:55
    at Array.forEach (<anonymous>)
    at Program:exit (/home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/rules/indent.js:1635:26)
    at /home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/jayen/docs/work/ridethewaverley/ride-the-waverley/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
```

Was not able to reproduce in codesandbox.

**Input Code**

```js
const foo = '';
require(`${foo}`)?.bar;
```

**Expected behavior**

Expected an assignment or function call and instead saw an expression. (no-unused-expressions)

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `eslintrc-that-CRA4-wont-find.json`

```js
{
  "extends": "react-app",
  "rules": {
    "array-bracket-spacing": [
      "error",
      "never"
    ],
    "comma-dangle": [
      "error",
      "always-multiline"
    ],
    "indent": [
      "error",
      2,
      {
        "CallExpression": {
          "arguments": "first"
        },
        "MemberExpression": "off",
        "flatTernaryExpressions": true,
        "ignoredNodes": [
          "JSXElement *"
        ]
      }
    ],
    "object-curly-spacing": [
      "error",
      "never"
    ],
    "react/jsx-indent-props": [
      "error",
      "first"
    ],
    "react/jsx-tag-spacing": [
      "error",
      {
        "beforeSelfClosing": "never",
        "beforeClosing": "never"
      }
    ],
    "react/self-closing-comp": [
      "error",
      {
        "component": true,
        "html": true
      }
    ],
    "sort-keys": [
      "error",
      "asc",
      {
        "caseSensitive": false,
        "natural": true
      }
    ],
    "sort-vars": "error"
  }
}
```

**Environment**

```
  System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
  Binaries:
    Node: 14.16.0 - /usr/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 6.14.11 - /usr/bin/npm
  npmPackages:
    @babel/eslint-parser: ^7.13.10 => 7.13.10 
    eslint: ^7.23.0 => 7.23.0 
```
- Babel version(s): [7.12.3]
- Node/npm version: [Node 14.16.0/npm 6.14.11]
- OS: [Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)]
- Monorepo: [no]
- How you are using Babel: [`eslint`]

**Possible Solution**

No but my workaround is to separate the require from the optional chaining.

**Additional context**

**Comment by JLHwung** [comment]
Do you have multiple copies of `@babel/types` in node_modules? See also https://github.com/babel/babel/issues/12985#issuecomment-796280417

**Comment by jayenashar** [comment]
i think i only have one copy

```
$ find -name types | grep @babel
./node_modules/@babel/types
```

```
├─┬ jscpd@3.3.25
│ └─┬ @jscpd/finder@3.3.25
│   └─┬ pug@3.0.2
│     └─┬ pug-code-gen@3.0.2
│       ├─┬ constantinople@4.0.1
│       │ └── @babel/types@7.13.13  deduped
│       └─┬ with@7.0.2
│         ├── @babel/types@7.13.13  deduped
│         └─┬ babel-walk@3.0.0-canary-5
│           └── @babel/types@7.13.13  deduped
└─┬ react-scripts@4.0.3
  ├─┬ @babel/core@7.12.3
  │ ├─┬ @babel/generator@7.13.9
  │ │ └── @babel/types@7.13.13  deduped
  │ ├─┬ @babel/helper-module-transforms@7.13.12
  │ │ ├─┬ @babel/helper-module-imports@7.13.12
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/helper-replace-supers@7.13.12
  │ │ │ ├─┬ @babel/helper-member-expression-to-functions@7.13.12
  │ │ │ │ └── @babel/types@7.13.13  deduped
  │ │ │ ├─┬ @babel/helper-optimise-call-expression@7.12.13
  │ │ │ │ └── @babel/types@7.13.13  deduped
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/helper-simple-access@7.13.12
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/helper-split-export-declaration@7.12.13
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ └── @babel/types@7.13.13  deduped
  │ ├─┬ @babel/helpers@7.13.10
  │ │ └── @babel/types@7.13.13  deduped
  │ ├─┬ @babel/template@7.12.13
  │ │ └── @babel/types@7.13.13  deduped
  │ ├─┬ @babel/traverse@7.13.13
  │ │ ├─┬ @babel/helper-function-name@7.12.13
  │ │ │ ├─┬ @babel/helper-get-function-arity@7.12.13
  │ │ │ │ └── @babel/types@7.13.13  deduped
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ └── @babel/types@7.13.13  deduped
  │ └── @babel/types@7.13.13 
  ├─┬ @svgr/webpack@5.5.0
  │ ├─┬ @babel/preset-env@7.13.12
  │ │ ├─┬ @babel/plugin-proposal-async-generator-functions@7.13.8
  │ │ │ └─┬ @babel/helper-remap-async-to-generator@7.13.0
  │ │ │   ├─┬ @babel/helper-wrap-function@7.13.0
  │ │ │   │ └── @babel/types@7.13.13  deduped
  │ │ │   └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/plugin-transform-classes@7.13.0
  │ │ │ └─┬ @babel/helper-annotate-as-pure@7.12.13
  │ │ │   └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/plugin-transform-exponentiation-operator@7.12.13
  │ │ │ └─┬ @babel/helper-builder-binary-assignment-operator-visitor@7.12.13
  │ │ │   ├─┬ @babel/helper-explode-assignable-expression@7.13.0
  │ │ │   │ └── @babel/types@7.13.13  deduped
  │ │ │   └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/plugin-transform-modules-systemjs@7.13.8
  │ │ │ └─┬ @babel/helper-hoist-variables@7.13.0
  │ │ │   └── @babel/types@7.13.13  deduped
  │ │ ├─┬ @babel/preset-modules@0.1.4
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ └── @babel/types@7.13.13  deduped
  │ ├─┬ @babel/preset-react@7.13.13
  │ │ └─┬ @babel/plugin-transform-react-jsx@7.13.12
  │ │   └── @babel/types@7.13.13  deduped
  │ └─┬ @svgr/plugin-jsx@5.5.0
  │   └─┬ @svgr/hast-util-to-babel-ast@5.5.0
  │     └── @babel/types@7.13.13  deduped
  ├─┬ babel-eslint@10.1.0
  │ └── @babel/types@7.13.13  deduped
  ├─┬ babel-jest@26.6.3
  │ ├─┬ @types/babel__core@7.1.14
  │ │ ├── @babel/types@7.13.13  deduped
  │ │ ├─┬ @types/babel__generator@7.6.2
  │ │ │ └── @babel/types@7.13.13  deduped
  │ │ └─┬ @types/babel__template@7.4.0
  │ │   └── @babel/types@7.13.13  deduped
  │ └─┬ babel-preset-jest@26.6.2
  │   └─┬ babel-plugin-jest-hoist@26.6.2
  │     └── @babel/types@7.13.13  deduped
  ├─┬ babel-preset-react-app@10.0.0
  │ ├─┬ @babel/plugin-proposal-optional-chaining@7.12.1
  │ │ └─┬ @babel/helper-skip-transparent-expression-wrappers@7.12.1
  │ │   └── @babel/types@7.13.13  deduped
  │ └─┬ @babel/preset-env@7.12.1
  │   └── @babel/types@7.13.13  deduped
  └─┬ jest-circus@26.6.0
    ├─┬ @types/babel__traverse@7.11.1
    │ └── @babel/types@7.13.13  deduped
    └─┬ jest-snapshot@26.6.2
      └── @babel/types@7.13.13  deduped
```

**Comment by nicolo-ribaudo** [comment]
> Was not able to reproduce in codesandbox.

Could you try copying your `package.json` and `package-lock.json` files in a separate repository, running `npm ci` to install exactly the same dependencies, and see if you can then reproduce it?

**Comment by jayenashar** [comment]
i did try various package versions and cleaning package-lock.json and node_modules.

i have now tried what you suggested.  can reproduce with the eslintrc above.  pushed to https://github.com/jayenashar/babel-issues-13060  running with `npx eslint --config eslintrc-that-CRA4-wont-find.json test.js `

**Comment by jayenashar** [comment]
may be worth pointing out that this works with either jscpd or source-map-explorer, but not both

**Comment by nicolo-ribaudo** [comment]
Thanks for the repro! It's 2 am here so I won't take a look at it now, but I'll do it in the next few days (unless @JLHwung does it first!).

**Comment by jayenashar** [comment]
thanks for telling me how to repro!

**Comment by fedeci** [comment]
I can look into this

**Comment by nicolo-ribaudo** [comment]
Oh thanks @fedeci!

**Comment by fedeci** [comment]
`@babel/parser` is parsing incorrectly `TemplateLiteral` quasis locations. [astexplorer](https://astexplorer.net/#/gist/4ae6763b6fb4efe5298b3420eeba0389/latest).

**Comment by nicolo-ribaudo** [comment]
@fedeci I'm not sure if it's just ASTExplorer that resets the config when sharing a link, but should enable the `estree` plugin (since that's what Babel does when used with ESLint).

**Comment by fedeci** [comment]
It is the same for `@babel/parser`, `@babel/eslint-parser` and `espree`. `@typescript-eslint/parser` is parsing it correctly.

**Comment by jayenashar** [comment]
based on https://github.com/babel/babel/issues/13060#issuecomment-808727006 and https://github.com/babel/babel/issues/13060#issuecomment-808824686 i suspect something is duplicated, but i can't spot it.  is there a way to get a list of all duplicates?

**Comment by JLHwung** [comment]
Thanks for the reproducing repo. This is https://github.com/facebook/create-react-app/issues/10636

The linting error is thrown from `babel-eslint@10`
```
npm ls babel-eslint
ionic-ride-the-waverley-app@0.0.1 /tmp/babel-issues-13060
└─┬ react-scripts@4.0.3
  └── babel-eslint@10.1.0
```
which is expected since `babel-eslint` 10 does not support optional chaining and had been deprecated.

@fedeci The position of template literal is adjusted in `@babel/eslint-parser` https://github.com/babel/babel/blob/5c0d2f603272add0164fb4dc0d880ae9d867f31b/eslint/babel-eslint-parser/src/convert/convertAST.js#L64-L80 so the position info on astexplorer does not match exactly to what we have in `@babel/eslint-parser`.

**Comment by fedeci** [comment]
That's strange, because both [`@babel/parser`](https://astexplorer.net/#/gist/ccba7f17823fc1d800b2b0baa9d50262/793c4fc97c2291285d041fa73b9d2b0caf372206) and [`@babel/eslint-parser`](https://astexplorer.net/#/gist/fd755884c7a5a0c94d4e22e8b89e890d/4baff2645d050c9906fc572bd0e3021e9c88f15e) parse the same exact location data🤔

**Comment by JLHwung** [comment]
> both @babel/parser and @babel/eslint-parser parse the same exact location data

https://github.com/babel/babel/blob/5c0d2f603272add0164fb4dc0d880ae9d867f31b/eslint/babel-eslint-parser/src/index.js#L46

Yeah, the `convertAST` is invoked _after_ the source is parsed by Babel. The location adjustment is required because Babel and Espree have different opinion on the location of template quasis.

**Comment by birdofpreyru** [comment]
As @fedeci suggests in his comment by the link above, the issue seems to be resolved in `@babel/eslint-parser@7.16.0` (at least, it does not manifest itself in my project, where I experienced it before).
